### PR TITLE
Update getting_started.md Document

### PR DESCRIPTION
**Pull request by jalev**
## Changes

Updates the `getting_started.md` to make it a little clearer in some places. 

## Verification
You read it a few times and if it makes sense then it has worked ;)

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -1,14 +1,12 @@
 # Getting started
 
-At its heart Thanos provides a global query view, data backup, and access historical data as core features. All three are well separated and Thanos deployments can be adapted to make use of each of them individually. This supports using just a subset of its features as well as a gradual rollout that immediately provides some of its benefits.
+Thanos provides a global query view, data backup, and historical data access as its core features in a single binary. All three features can be run independently of each other. This allows you a subset of Thanos features for immediate benefit or testing, while also making it flexible for gradual roll outs in more complex environments.
```
Comment (Bplotka) [inline docs/getting_started.md]: `This allows you a subset`? 

Maybe `This allows you to have a subset..`?

File: docs/getting_started.md [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
+Thanos provides a global query view, data backup, and historical data access as its core features in a single binary. All three features can be run independently of each other. This allows you a subset of Thanos features for immediate benefit or testing, while also making it flexible for gradual roll outs in more complex environments. 
 
-The following examples configure Thanos to work against a Google Cloud Storage bucket. However, any object storage (S3, HDFS, DigitalOcean Spaces, ...) can be used by using the equivalent flags to connect to the S3 API.
-
-See [this](storage.md) for up-to-date list of available object stores for Thanos.
+In this quick-start guide, we will configure Thanos and all components mentioned to work against a Google Cloud Storage bucket. Thanos is able to use [many other object stores (S3, S3 API, HDFS, DigitalOcean Spaces, ...)](storage.md), and you can substitute Google Cloud specific flags in this guide with those of your object store detailed in the [Storage document](storage.md).
```
Comment (Bplotka) [inline docs/getting_started.md]: This is not entirely true, as we have clients only to S3 and GCS (and maybe minio)

Comment (Bplotka) [reply docs/getting_started.md]: also you missed GCS (:

File: docs/getting_started.md [hunk 3]
```diff
@@ -49,65 +46,69 @@ If you are not interested in backing up any data, the `--gcs.bucket` flag can si
 * _[Example Kubernetes manifest with GCS upload](../kube/manifests/prometheus-gcs.yaml)_
 * _[Details & Config for other object stores](./storage.md)_
 
-### Query Access
+### [Store API](components/store.md)
+
+The Sidecar comes with a [gRPC](https://grpc.io/) API called the _[Store API](components/store.md)_. The Store API allows you to query metric data in Prometheus, and data backed up into the Object Store bucket.
 
-Thanos comes with a highly efficient gRPC-based Store API for metric data access across all its components. The sidecar implements it in front of its connected Prometheus server. While it is ready to use with the above example, we must additionally configure the sidecar to join a Thanos cluster.
+In order to expose the Store API, we also need to put the Sidecar into '_cluster_' mode. A cluster can be made up of one or more Sidecars. Clustering uses the [Gossip Protocol](https://en.wikipedia.org/wiki/Gossip_protocol) to allow Thanos components- including the Sidecar itself- to be able to find other Sidecars. Clustering is explained in greater detail in the [Communication Between Components](#communication-between-components) section.
```
Comment (Bplotka) [inline docs/getting_started.md]: This section is entirely false. There is literally NO clustering needed for sidecars, and we have to fix this confusion ASAP. In fact we will get rid of gossip as this only causes problems: https://github.com/improbable-eng/thanos/issues/484

Comment (Bplotka) [reply docs/getting_started.md]: Whole Communication between components needs to be revisited as well ):

Comment (Bplotka) [reply docs/getting_started.md]: TL;DR only querier needs to know where storeAPIs are (sidecars, rulers, store gateways or other qieriers)

File: docs/getting_started.md [hunk 4]
```diff
@@ -49,65 +46,69 @@ If you are not interested in backing up any data, the `--gcs.bucket` flag can si
...
+    --http-address              0.0.0.0:19191 \            # HTTP endpoint for collecting metrics on the Sidecar
+    --cluster.address           0.0.0.0:19391 \            # Gossip endpoint used to expose Gossip metadata for the current node
+    --cluster.advertise-address 127.0.0.1:19391 \          # Gossip endpoint at which the node advertises itself at to other members of the cluster
+    --cluster.peers             127.0.0.1:19391 \          # Static list of peers where the node will get info about the cluster
 ```
 
 * _[Example Kubernetes manifest](../kube/manifests/prometheus.yaml)_
 * _[Example Kubernetes manifest with GCS upload](../kube/manifests/prometheus-gcs.yaml)_
 
-### External Labels
```
Comment (Bplotka) [inline docs/getting_started.md]: I would vote for leaving it here as there are essential and not _only_ for deduplication!

File: docs/getting_started.md (hunk 3 above)
Comment (jalev) [reply docs/getting_started.md]: Ah! Thanks for the clarification. I was going to get to the Component comms later today but it looks like I need to revisit the entire section here. 🤔

File: docs/getting_started.md (hunk 4 above)
Comment (jalev) [reply docs/getting_started.md]: kk, I'll put it back.

File: docs/getting_started.md (hunk 1 above)
Comment (jalev) [reply docs/getting_started.md]: 👍

File: docs/getting_started.md (hunk 2 above)
Comment (jalev) [reply docs/getting_started.md]: Ah sugar, good spot.

File: docs/getting_started.md (hunk 4 above)
Comment (jalev) [reply docs/getting_started.md]: Could you explain a little what else it's used for? I saw in the source that it has a role in the data backup process but I'm a little unclear.

File: docs/getting_started.md [hunk 5]
```diff
@@ -24,19 +23,18 @@ make
 
 The `thanos` binary should now be in your `$PATH` and is the only thing required to deploy any of its components.
 
-## Sidecars
+## [Sidecar](components/sidecar.md)
 
-Thanos integrates with existing Prometheus servers through a sidecar process, which runs on same machine/in the same pod as the Prometheus server itself. It only works with Prometheus instances of version 2.0.
+Thanos integrates with existing Prometheus (v2.2.1+) servers through a [Sidecar process](https://docs.microsoft.com/en-us/azure/architecture/patterns/sidecar#solution), which runs on the same machine or in the same pod as the Prometheus server/s.
```
Comment (Bplotka) [inline docs/getting_started.md]: `/s. ` is not needed? sidecar will run in the same pod as Prometheus server - no way we can have more than 1 Prom container in pod... and that `/s` suggests isn't it?

Comment (jalev) [reply docs/getting_started.md]: True yeah. I'll change it

**Comment by jalev** [comment]
Ready to roll. Thanks @Bplotka 👍

**Comment by Bplotka** [comment]
Thanks!

**Pull request merged**